src/client/Microsoft.Identity.Client/Utils/StopWatchService.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.Utils
```

Comment:
implementation from MISE with slight modifications
https://identitydivision.visualstudio.com/DevEx/_git/MISE?version=GBgeok/telemetry-exporter&path=/src/Caching/Utility.cs

Reply:
Does MISE have any unit tests for this?

Reply:
They do but I dont think they run them because of the inconsistency of the performance of the build agents.